### PR TITLE
Scripts for Jenkins and Devilbox

### PR DESCRIPTION
**Pull request by rjcf18**
- Project base initiated with composer file 
- Created the scripts for the generation of the dockerfiles given the permutation of Joomla! CMS branches and php versions (php versions used an xml file as input with the available versions)

### CODE REVIEW

File: src/DockerfilesGenerator.php [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+				$dockerfile .= "\tgit clone --depth 1 -b " . $branch . " --single-branch ";
+				$dockerfile .= "https://github.com/joomla/joomla-cms.git /usr/src/joomla ; \\\n";
+				$dockerfile .= "\tchown -R www-data:www-data /usr/src/joomla\n\n";
+				$dockerfile .= "# Copy init scripts and custom .htaccess\n";
+				$dockerfile .= "COPY docker-entrypoint.sh /entrypoint.sh\n";
+				$dockerfile .= "COPY makedb.php /makedb.php\n\n";
+				$dockerfile .= "RUN echo 'ServerName localhost' >> /etc/apache2/apache2.conf\n\n";
+				$dockerfile .= "ENTRYPOINT [\"/entrypoint.sh\"]\n";
+				$dockerfile .= "CMD [\"apache2-foreground\"]";
+
```
Comment (nibra) [inline src/DockerfilesGenerator.php]: Move this to a separate file, so it is easier to edit.

Comment (rjcf18) [inline src/DockerfilesGenerator.php]: Ok so move it to a separate file and just replace the php version and branch in the right place right?

Comment (nibra) [inline src/DockerfilesGenerator.php]: Yes, exactly.

File: bash_scripts/configuration.php [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+	public $gzip = '0';
+	public $error_reporting = 'default';
+	public $helpurl = 'https://help.joomla.org/proxy?keyref=Help{major}{minor}:{keyref}&lang={langcode}';
+	public $ftp_host = '';
+	public $ftp_port = '';
+	public $ftp_user = '';
+	public $ftp_pass = '';
+	public $ftp_root = '';
+	public $ftp_enable = '0';
+	public $tmp_path = '/shared/httpd/${INSTANCEFOLDER}/htdocs/administrator/logs';
```
Comment (jatitoam) [inline bash_scripts/configuration.php]: these paths are a bit specific in the `/shared/httpd` path, or is that something of devilbox?

File: RoboFile.php [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+/**
+ * This is project's console commands configuration for Robo task runner.
+ *
+ * @see  http://robo.li/
+ */
+
+require __DIR__ . '/vendor/autoload.php';
+
+
+class RoboFile extends \Robo\Tasks
```
Comment (rjcf18) [inline RoboFile.php]: Yes, for now just bash but I'm planning on moving bash scripts to Robo later on,

File: bash_scripts/configuration.php (hunk 2 above)
Comment (rjcf18) [inline bash_scripts/configuration.php]: yes that path is inside the php containers in devilbox, the shared/httpd/ folder is where the joomla instance folders will be.

**Pull request merged**